[Shannon Tsou] iP

tllshan opened the pull request:
# ChatterChicken
>"Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

ChatterChicken frees your mind of having to remember things you need to do. It's

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/tllshan/ip/archive/refs/tags/A-Jar.zip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😄

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tllshan committed:
Implement Level 0. Rename, Greet, Exit

Renamed Duke to ChatterChicken.
Added greeting and exit.

tllshan committed:
Implemented Level 0. Rename, Greet, Exit

Renamed Duke to ChatterChicken.
Added greeting and exit.

tllshan committed:
Level 1. Echo

Added indentation to lines printed.
Added scanner to echo user input.
Exits on "bye" input.

tllshan committed:
Implement Level 1. Echo

Added indentation to lines printed.
Added scanner to echo user input.
Exits on "bye" input.

tllshan committed:
Level 2. Add, List

Added Task and List classes.
Adds user inputs as tasks to list.
Prints list on "list" input.

tllshan committed:
Implement Level 2. Add, List

Added Task and List classes.
Adds user inputs as tasks to list.
Prints list on "list" input.

tllshan committed:
Implement Level 3. Mark as Done

Added isDone field in Task.
Added mechanism to mark and unmark tasks.

tllshan committed:
Level 3. Mark as Done

Added isDone field in Task.
Added mechanism to mark and unmark tasks.

tllshan committed:
Level 4. ToDos, Events, Deadlines

Task is an interface.
ToDo, Deadline and Event implement Task.

tllshan committed:
Level 4. ToDos, Events, Deadlines

Task is an interface.
ToDo, Deadline and Event implement Task.

tllshan committed:
Implement Level 4. ToDos, Events, Deadlines

Task is an interface.
ToDo, Deadline and Event implement Task.

tllshan committed:
A-TextUiTesting

Tests if ACTUAL.TXT matches EXPECTED.TXT for given inputs in input.txt

tllshan committed:
A-TextUiTesting

Tests if ACTUAL.TXT matches EXPECTED.TXT for given inputs in input.txt

tllshan committed:
Implement A-TextUiTesting

Tests if ACTUAL.TXT matches EXPECTED.TXT for given inputs in input.txt

tllshan committed:
Level 5. Handle Errors

Implemented error handling for faulty inputs.
Changed Task from interface to abstract class.

tllshan committed:
Level 5. Handle Errors

Implemented error handling for faulty inputs.
Changed Task from interface to abstract class.

tllshan committed:
Implement Level 5. Handle Errors

Implemented error handling for faulty inputs.
Changed Task from interface to abstract class.

tllshan committed:
Level 6. Delete

Added delete task function to list.

tllshan committed:
Level 6. Delete

Added delete task function to list.

tllshan committed:
Implement Level 6. Delete

Added delete task function to list.

tllshan committed:
Add javadocs for classes and methods

tllshan committed:
Add javadocs for classes and methods

tllshan committed:
Add javadocs for classes and methods

tllshan committed:
Implement Level-7 Save for ToDo

tllshan committed:
Implement Level-7 Save for ToDo

tllshan committed:
Implement Level-7 Save for ToDo

tllshan committed:
Implement Level-7 Save for Deadline and Event

tllshan committed:
Implement Level-7 Save for Deadline and Event

tllshan committed:
Implement Level-7 Save for Deadline and Event

tllshan committed:
Merge pull request #1 from tllshan/branch-Level-7

Branch level 7

tllshan committed:
Implement Level-8 Dates

tllshan committed:
Implement Level-8 Dates

tllshan committed:
Implement Level-8 Dates

tllshan committed:
Implement A-MoreOOP: Storage and Parser

tllshan committed:
Implement A-MoreOOP: Storage and Parser

tllshan committed:
Implement A-MoreOOP: Command class

tllshan committed:
Implement A-MoreOOP: Command class

tllshan committed:
Fix bug in Storage class for saving and loading mechanism

tllshan committed:
Implemented A-MoreOOP: Ui class

tllshan committed:
Implemented A-Packages

tllshan committed:
Update runtest.bat for compatibility with package structure

imkwokyong commented on src/main/java/chatterchicken/parser/Parser.java:
```diff
@@ -0,0 +1,148 @@
+            case "event":
+                task = parseEvent(input);
+                break;
+        }
```
Shouldn't there not be indentation in the `case` clauses?

imkwokyong commented on src/main/java/chatterchicken/ChatterChicken.java:
```diff
@@ -0,0 +1,74 @@
+import java.util.NoSuchElementException;
```
`package chatterchicken;` is missing
```suggestion
package chatterchicken;

import java.util.NoSuchElementException;
```

imkwokyong commented on src/main/java/chatterchicken/ui/Ui.java:
```diff
@@ -0,0 +1,93 @@
+
+    private static final String LINE = "\n    _____________________________________________________________________________\n";
+
+    private static final String INDENT = "      ";
+    private static final String INDENT_BIG = "        ";
```
I like how you differentiate the indentations and make them constants, it makes reading the output messages much easier!

imkwokyong commented on src/main/java/chatterchicken/tasklist/TaskList.java:
```diff
@@ -0,0 +1,126 @@
+    /**
+     * Prints the list of tasks with their respective indexes.
+     */
+    public void printList() {
+        System.out.println(ChatterChicken.LINE + ChatterChicken.INDENT + "Here are the tasks in your list:");
+        for (int i = 0; i < taskList.size(); i++) {
+            System.out.println(ChatterChicken.INDENT_BIG + (i + 1) + "." + taskList.get(i).getTaskForPrinting());
+        }
+        System.out.println(ChatterChicken.LINE);
+    }
```
Perhaps this could be delegated to the UI?

imkwokyong commented on src/main/java/chatterchicken/task/Task.java:
```diff
@@ -0,0 +1,40 @@
+        this.isDone = isDone;
+    }
+
+    protected String checkDone() {
```
Perhaps a more intuitive name to show that it returns a String?

imkwokyong commented on src/main/java/chatterchicken/tasklist/TaskList.java:
```diff
@@ -0,0 +1,126 @@
+        try {
+            Task task = taskList.get(getIndex(input));
+            task.setDone(true);
+            System.out.println(ChatterChicken.LINE
+                    + ChatterChicken.INDENT + "Nice! I've marked this task as done:\n"
+                    + ChatterChicken.INDENT_BIG + task.getTaskForPrinting()
+                    + ChatterChicken.LINE);
```
Perhaps delegating the printing to the UI will be better? 
I also suggest this for other methods in the TaskList.java.

imkwokyong reviewed (commented):
I think there are some spacing inconsistencies with brackets and indentations, so it's good to check on them.
Otherwise, good job so far! You got this!

tllshan committed:
Fix bugs in all classes

tllshan committed:
Restructure packages

tllshan committed:
Implement A-JUnit for parser and tasklist

tllshan committed:
Create MANIFEST.MF

tllshan committed:
Add A-CodingStandard

tllshan committed:
Add A-JavaDoc

tllshan committed:
Implement Level 9

tllshan committed:
Merge pull request #4 from tllshan/branch-A-CodingStandard

Add A-CodingStandard

tllshan committed:
Merge pull request #5 from tllshan/branch-A-JavaDoc

Add A-JavaDoc

tllshan committed:
Merge branch 'master' into branch-Level-9

tllshan committed:
Merge pull request #6 from tllshan/branch-Level-9

Implement Level 9

neyapraveen commented on src/main/java/chatterchicken/ChatterChicken.java:
```diff
@@ -57,35 +66,36 @@ private void run() {
+            tasks.addTask(parser.parseTask(action, taskDescription));
+            break;
+        default:
+            throw new CCException("OOPS!!! I'm sorry, but I don't know what that means :<");
```
Good job with changing the indentation!

neyapraveen commented on src/main/java/chatterchicken/tasklist/TaskList.java:
```diff
@@ -93,6 +93,16 @@ public void deleteTask(String input) throws CCException {
+    public void find(String input) {
+        ArrayList<Task> matchingTasks = new ArrayList<>();
+        for (Task task : taskList) {
+            if(task.getTaskDescription().contains(input)) {
```
Should there be a space here?

neyapraveen commented on src/main/java/chatterchicken/parser/Parser.java:
```diff
@@ -104,17 +115,17 @@ private Deadline parseDeadline(String taskDescription) throws CCException {
             throw new CCException("OOPS!!! Incorrect format for deadline.");
```
Could we be clearer by specifying the required format?

neyapraveen commented on src/main/java/chatterchicken/ui/Ui.java:
```diff
@@ -5,23 +5,25 @@
+ */
 public class Ui {
 
     private static final String LINE = "\n    _____________________________________________________________________________\n";
```
Good job with using all uppercase for the constants

neyapraveen reviewed (commented):
LGTM overall! Follows coding standard well, just a few nits to fix.

tllshan committed:
Import javafx

tllshan committed:
Create gui class

tllshan committed:
Implement main and mainwindow classes

tllshan committed:
Add FXML

tllshan committed:
Add FXML

tllshan committed:
Merge pull request #8 from tllshan/branch-Level-10

Branch level 10

tllshan committed:
Update README.md

tllshan committed:
Generate new JAR

tllshan committed:
Implement A-CodeQuality

tllshan committed:
Implement A-Assertions

tllshan committed:
Merge pull request #9 from tllshan/branch-A-Assertions

Implement A-Assertions

tllshan committed:
Merge pull request #10 from tllshan/branch-A-CodeQuality

Implement A-CodeQuality

tllshan committed:
Implement BCD-Extension

Duke does not provide statistics on the tasks in the list.

Users do not get a statistical overview of their tasks.

Adding functionality to show task list statistics allows users to get more insight into their tasks.

Let's add methods in TaskList that display
*the number of completed tasks
*the number of uncompleted tasks
*the number of each type of task

tllshan committed:
Break down Command class into separate classes

The Command class is an all-purpose class that encompasses all valid commands. The main execution of the commands is handled by the executeCommand method in the Duke class and the corresponding methods in TaskList.

Having specific classes for each command will allow the unique execution processes of each command to be decentralised.

Let's make Command be an abstract parent class with the execute method, and make individual classes for each valid command.

This means that all Commands can be run by Duke using the execute method, without having to know the details of how each command is being executed.

tllshan committed:
Merge branch 'master' into branch-Level-7

tllshan committed:
Merge pull request #11 from tllshan/branch-Level-7

Branch level 7

tllshan committed:
Merge branch 'master' into branch-Level-8

tllshan committed:
Merge pull request #12 from tllshan/branch-Level-8

Branch level 8

tllshan committed:
Rename to DukeException

tllshan committed:
Implement A-BetterGUI

tllshan committed:
Remove redundant Ui class

tllshan committed:
Update Ui

tllshan committed:
Add user guide

tllshan committed:
Fix bug in stats

tllshan committed:
Update Ui

tllshan committed:
Update user guide

tllshan committed:
Update user guide

tllshan committed:
Fix bug in jar file for linux
